packages/@aws-cdk/core/test/nested-stack.test.ts

```diff
@@ -31,4 +33,20 @@ describe('nested-stack', () => {
 
     expect(nestedStack.templateOptions.description).toEqual(description);
   });
+
+  test('a nested-stack has shorter logical id', () => {
+    // GIVEN
+    const shortlogicalId = 'MyNestedStackNestsMyNestedStack47285AF6';
+    const shortlogicalIdFlag = { [cxapi.ENABLE_SHORTER_LOGICAL_ID_NESTED_STACKS]: true };
+    const app = new App({
+      context: shortlogicalIdFlag,
+    });
+
+    const stack = new Stack(app);
+    const nestedStack = new NestedStack(stack, 'MyNestedStack');
+    const cfn_nestedStack = (nestedStack.node.defaultChild) as CfnStack;
```

Comment:
inspecting the cfn_nestedStack object, it has `logicalId: '${Token[Default.MyNestedStack.Nests.MyNestedStack.LogicalID.15]}'` , what are the various parts of this and why was it implemented in this way?

Reply:
I think the issue here is that it wasn't intended to be this way. Since it is a bug, this task probably needs a deep dive to figure out why we ended up here.